akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2StreamHandling.scala

```diff
@@ -332,6 +498,132 @@ private[http2] trait Http2StreamHandling { self: GraphStageLogic with LogHelper
     def shutdown(): Unit = outlet.fail(Http2StreamHandling.ConnectionWasAbortedException)
   }
 
+  trait OutStream {
+    def canSend: Boolean
+    def cancelStream(): Unit
+    def endStreamIfPossible(): Option[FrameEvent]
+    def nextFrame(maxBytesToSend: Int): DataFrame
+    def increaseWindow(delta: Int): Unit
+    def isDone: Boolean
+  }
+  object OutStream {
+    def apply(sub: Http2SubStream): OutStream = {
+      val subIn = new SubSinkInlet[Any](s"substream-in-${sub.streamId}")
+      val info = new OutStreamImpl(sub.streamId, None, multiplexer.currentInitialWindow)
+      info.registerIncomingData(subIn)
+      sub.data.runWith(subIn.sink)(subFusingMaterializer)
+      info
+    }
+  }
+  class OutStreamImpl(
```

Comment:
Carried over as much unchanged as possible from Http2Multiplexer. We can clean this up later. The important bit is that we avoid side-effects crossing between stream handling and multiplexer in an uncontrolled way.